java/src/main/java/com/adhoc/flightsql/DremioFlightSqlClientDemoApp.java

```diff
@@ -0,0 +1,122 @@
+
+  public DremioFlightSqlClientDemoApp(BufferAllocator bufferAllocator) {
+    super(bufferAllocator);
+  }
+
+  public static void main(final String[] args) throws Exception {
+    final Options options = new Options();
+
+    options.addOption("host", "host", true, "Host to connect to");
+    options.addOption("port", "port", true, "Port to connect to");
+    options.addOption("username", "username", true, "Auth username");
+    options.addOption("password", "password", true, "Auth password");
+    options.addRequiredOption("command", "command", true, "Method to run");
+
+    options.addOption("query", "query", true, "Query");
+    options.addOption("catalog", "catalog", true, "Catalog");
+    options.addOption("schema", "schema", true, "Schema");
+    options.addOption("table", "table", true, "Table");
+
+    CommandLineParser parser = new DefaultParser();
+    HelpFormatter formatter = new HelpFormatter();
+    CommandLine cmd;
+
+    try {
+      cmd = parser.parse(options, args);
+      try (final DremioFlightSqlClientDemoApp thisApp = new DremioFlightSqlClientDemoApp(
+          new RootAllocator(Integer.MAX_VALUE))) {
+        thisApp.executeApp(cmd);
+      }
+
+    } catch (final ParseException e) {
+      System.out.println(e.getMessage());
+      formatter.printHelp("DremioFlightSqlClientDemoApp -host localhost -port 32010 ...", options);
+      throw e;
+    }
+  }
+
+  /**
+   * Adds a {@link CallOption} to the current {@code callOptions} array.
+   */
+  public void addCallOption(final CallOption optionToAdd) {
+    callOptions.add(optionToAdd);
+  }
+
+  /**
+   * Calls {@link DremioFlightSqlClientDemoApp#createFlightSqlClient(String, String, String, String)}
+   * in order to create a {@link FlightSqlClient} to be used in future calls,
+   * and then calls {@link DremioFlightSqlClientDemoApp#executeCommand(CommandLine)}
+   * to execute the command parsed at execution.
+   *
+   * @param cmd Parsed {@link CommandLine}; often the result of {@link DefaultParser#parse(Options, String[])}.
+   */
+  public void executeApp(CommandLine cmd) throws Exception {
+    createFlightSqlClient(
+        cmd.getOptionValue("host", "localhost").trim(), cmd.getOptionValue("port", "32010").trim(),
+        cmd.getOptionValue("username", "dremio").trim(), cmd.getOptionValue("password", "dremio123").trim());
+
+    executeCommand(cmd);
+  }
+
+  /**
+   * Creates a {@link FlightSqlClient} to be used with the example methods.
+   *
+   * @param host client's hostname.
+   * @param port client's port.
+   * @param user client's username auth.
+   * @param pass client's password auth.
+   */
+  public void createFlightSqlClient(final String host, final String port, final String user, final String pass) {
+    final ClientIncomingAuthHeaderMiddleware.Factory factory =
+        new ClientIncomingAuthHeaderMiddleware.Factory(new ClientBearerHeaderHandler());
+    final FlightClient client = FlightClient.builder()
```

Comment:
This should have the Client cookie middleware (since this is an example).

Reply:
Done